### ISSUE TITLE
Auto Connection fail on VCX and new agent support

### ISSUE THREAD

**Issue by khujin1** [openwallet-foundation/owl-agent-test-harness#110]
During ACA-PY and VCX test, I found the following error.
Agents with auto connection setting show an error in the current test code.
(I found same issue on aca-py `--auto-accept-invites` setting.)
Maybe we need to change `0160-connection.py` for auto-connection features?

`./manage run -d acapy -b vcx -t @AcceptanceTest -t ~@wip`

      And "Faber" and "Bob" have an existing connection  features/steps/0160-connection.py:245 1.862s
      Assertion Failed: FAILED SUB-STEP: And "Bob" receives the connection invitation
      Substep info: Traceback (most recent call last):
        File "/usr/local/lib/python3.7/site-packages/behave/model.py", line 1329, in run
          match.run(runner.context)
        File "/usr/local/lib/python3.7/site-packages/behave/matchers.py", line 98, in run
          self.func(context, *args, **kwargs)
        File "features/steps/0160-connection.py", line 113, in step_impl
          assert expected_agent_state(invitee_url, "connection", context.connection_id_dict[invitee][context.inviter_name], "invited")
      AssertionError
      Traceback (of failed substep):
        File "/usr/local/lib/python3.7/site-packages/behave/model.py", line 1329, in run
          match.run(runner.context)
        File "/usr/local/lib/python3.7/site-packages/behave/matchers.py", line 98, in run
          self.func(context, *args, **kwargs)
        File "features/steps/0160-connection.py", line 113, in step_impl
          assert expected_agent_state(invitee_url, "connection", context.connection_id_dict[invitee][context.inviter_name], "invited")

      Captured stdout:
      From http://0.0.0.0:9030 Expected state ['invited'] but received request , with a response status of 200

And I create new branch for `vcxagencynode`(agent repo : https://github.com/AbsaOSS/vcxagencynode)
https://github.com/sktston/aries-agent-test-harness/tree/nodeagency
this repo includes node agency back-channel and vcx support.(but same issue on connection test)

I still have several issues(connection, change SDK to Aries-VCX, etc.), but do you need to merge in Master?

**Comment by ianco** [comment]
Hi @khujin1 thanks for creating the backchannel for the new VCX agent!  I suggest you open a draft pull request to the Aries Agent Test Harness repository and we can take a look.  FYI @nodlesh has been doing a lot of work matching up the test scenarios to the Aries RFC's so he can probably answer questions about specific tests.  Let us know if you need any pointers to documentation about backchannels etc.

**Comment by nodlesh** [comment]
Hi @khujin1 I also want to say thanks for creating that VCX backchannel, along with thanks for using the Aries Agent Test Harness.  
On the connection tests, they strictly follow what is set out in the [RFC](https://github.com/hyperledger/aries-rfcs/tree/master/features/0160-connection-protocol). There is no talk of, or if so vaguely, about an auto accept. So the tests don't know anything or do anything in an auto accept invites scenario. 
There are a few things we can consider here. 
First, can you run your VCX agent without the auto accept invites or is this the base behaviour? 
Second, we could make provisions in the connection tests for auto accept either through command line parameters, or test tags in feature files.  
Thirdly we could make the backchannels return what is expected by the tests on connection when the auto accept is on, it would be faking out the test, but if the backchannel knows it is on auto accept then the net result would be the same. 
There may be other ways to accomplish this. I will post here if I think of more. Hopefully the First thing I mentioned above can get us there. 
Looking forward to working through this with you. 
Cheers!

**Comment by khujin1** [comment]
Hi @nodlesh @ianco.  Thank you for your comment.

I already know that RFC is ambiguous about auto connection, but most of the others will use auto connection for convenience.

I already tried to find the auto accept option in vcx-agency, but I couldn't do it unless I modified the code. this might be the base behaviour. but I will check again with patrik(https://github.com/Patrik-Stas) who in charge of vcxagencynode.

I will try the rest of the suggestions and let you know.

When the back-channel is completed, I will create a draft PR. 
Thanks again.
